compose/ui/ui-test-junit4/src/skikoTest/kotlin/androidx/compose/ui/test/ComposeUiSkikoTestTest.kt

```diff
@@ -429,6 +427,33 @@ class ComposeUiSkikoTestTest {
         onNodeWithTag("test").performTextClearance()
         assertThat(text.text).isEqualTo("")
     }
+
+    // See https://github.com/JetBrains/compose-multiplatform/issues/3117
+    @Test
+    fun recompositionCompletesBeforeSetContentReturns() = repeat(300) {
```

Comment:
I tried to run this test as is but without the fix - "isApplyObserverNotificationPending" to reproduce the issue. It didn't fail on my linux machine - I made ~10 attempts. Then I increased the repeat to 3000 and it fails on each run attempt after ~1k iterations.With the fix it works fine with 3k iterations.

Did it fail for your with 300 iterations? Is it worth to increase the number of iterations to make this test less flaky in case `isIdle` implementation changes for some reason later?

Reply:
It fails pretty consistently for me at 300, but let's make it 1000 just in case.